cli/server/server.go

```diff
@@ -259,24 +262,43 @@ func restoreDB(ctx *cli.Context) error {
 		chain.Close()
 	}()
 
-	var start uint32
-	if ctx.Bool("incremental") {
-		start = reader.ReadU32LE()
-		if chain.BlockHeight()+1 < start {
-			return cli.Exit(fmt.Errorf("expected height: %d, dump starts at %d",
-				chain.BlockHeight()+1, start), 1)
-		}
+	var start = reader.ReadU32LE()
+	if reader.Err != nil {
+		return cli.Exit(err, 1)
+	}
+	var allBlocks = reader.ReadU32LE()
+	if reader.Err != nil {
+		return cli.Exit(err, 1)
+	}
+	var versionOrLen = reader.ReadU32LE()
+	if reader.Err != nil {
+		return cli.Exit(err, 1)
+	}
```

Comment:
You don't need to read the third integer. You should read only first two integers, the second one is either the number of blocks in the file or size of the block bytes (previously we discussed that it's a block version, but it's a mistake: serialized block starts with the size).